#!/usr/bin/env python3
PACKAGE = "strategy"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#------------------------------------------------------------------------------------
# Game Start
#------------------------------------------------------------------------------------
gen.add("Game_start",    bool_t,    0, "A Boolen parameter",  False)

#------------------------------------------------------------------------------------
# Robot mode
#------------------------------------------------------------------------------------
mode_enum = gen.enum([ gen.const("Idle",         str_t, "idle",         "robot is at ide mode"),
                       gen.const("Setting",      str_t, "Setting",      "Setting the robot"),
                       gen.const("Navigating",   str_t, "Navigating",   "the robot start to navagate"),
                       gen.const("Test",  str_t, "test",  "test the robot route"), 
                      #  gen.const("QGA",  str_t, "QGA",  "calculate the robot route"),
                       gen.const("Service",  str_t, "Service",  "Servie the customer")], "An enum to set mode")
gen.add("Robot_mode", str_t, 0, "Select Game mode", "idle", edit_method = mode_enum)



#------------------------------------------------------------------------------------
# item_list
#------------------------------------------------------------------------------------
item_enum = gen.enum([ gen.const("initial",    str_t, "initial",         ""),
                       gen.const("shelf",   str_t, "shelf",        ""),
                      #  gen.const("item2",   str_t, "item2",        ""), 
                      #  gen.const("item3",   str_t, "item3",        ""),
                       gen.const("Table1",   str_t, "Table1",        ""),
                       gen.const("Table2",   str_t, "Table2",        "")], "An item enum")
gen.add("Item", str_t, 0, "Select the item u want to go", "initial", edit_method = item_enum)

#------------------------------------------------------------------------------------
# setting option
#------------------------------------------------------------------------------------
gen.add("Get_loc",    bool_t,    0, "A Boolen parameter",  False)
gen.add("Reset_loc",     bool_t,    0, "A Boolen parameter",  False)
#------------------------------------------------------------------------------------
# nav_option
#------------------------------------------------------------------------------------
nav_mode_enum = gen.enum([  gen.const("test",        str_t, "test",             ""),
                       gen.const("directory",         str_t, "directory",         "")], "An naav mode enum")
gen.add("Nav_mode", str_t, 0, "Select the nav mode", "directory", edit_method = nav_mode_enum)
gen.add("Nav_start",    bool_t,    0, "A Boolen parameter",  False)
gen.add("Nav_stop",    bool_t,    0, "A Boolen parameter",  False)
gen.add("SaveYaml",    bool_t,    0, "A Boolen parameter",  False)
gen.add("LoadYaml",    bool_t,    0, "A Boolen parameter",  False)
exit(gen.generate(PACKAGE, "strategy", "Robot"))






# gen.add("game_start",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)



